### PR TITLE
004 Script for requirement [PerformAudioPassThru] SDL must transfer reque…

### PR DESCRIPTION
**Pull request by maria-mih**
…st to HMI in case of valid audioPassThruIcon param

All common functions used available at #703
Testing coverage available at #643

@istoimenova Please review

### CODE REVIEW

**Comment by istoimenova** [comment]
@maria-mih, please reference script in the TestSet.

**Comment by istoimenova** [comment]
@maria-mih, the file is put in folder sdl_atf_test_scripts_open/test_scripts/PerformAudioPassThru/, should be in sdl_atf_test_scripts_open/test_scripts/API/PerformAudioPassThru/

File: test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+-- as well as another related to request valid mandatory params
+-- SDL must transfer UI.PerformAudioPassThru (<audioPassThruIcon>, other params)_request + Speak_request (depends on parameters provided by the app) to HMI
+--1. Used preconditions
+--1. Only mandatory parameters are present and within bounds (samplingRate, maxDuration, bitsPerSample, audioType)
+--2. AudioPassThruIcon exists at apps sub-directory of AppStorageFolder (value from ini file)
+--2. Performed steps
+-- Send PerformAudioPassThru (audioPassThruIcon, mandatory params) from mobile to SDL and check:
+--2.1 SDL sends UI.PerformAudioPassThru (audioPassThruIcon, mandatory params) to HMI
+--2.2 SDL sends TTS.Speak to HMI
+--2.3 HMI sends UI.PerformAudioPassThru (SUCCESS) to SDL
```
Comment (istoimenova) [inline test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua]: @maria-mih, points 2.3-2.4 are redundant in description.

File: test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+--2.4 HMI sends TTS.Speak (SUCCESS) to SDL
+-- Expected result:
```
Comment (istoimenova) [inline test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua]: istoimenova 3 minutes ago Member Pending

@maria-mih , expected result should be:
SDL sends UI.PerformAudioPassThru (audioPassThruIcon, other params) to HMI
SDL sends TTS.Speak to HMI

File: test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+-- SDL sends PerformAudioPassThru (SUCCESS) to mobile app
+
+---------------------------------------------------------------------------------------------
+
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+
+--[[ Required Shared libraries ]]
+local commonFunctions = require ('user_modules/shared_testcases/commonFunctions')
```
Comment (istoimenova) [inline test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua]: @maria-mih, please remove space before brackets, use this for all functions that are called / required.

File: test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+commonSteps:PutFile("Precondition_PutFile_With_Icon","icon.png")
+
+function Test:Precondition_Check_audioPassThruIcon_Existence()
+  testCasesForPerformAudioPassThru:Check_audioPassThruIcon_Existence(self)
+end
+
+function Test:Precondition_ActivateApp()
+  testCasesForPerformAudioPassThru:ActivateAppDiffPolicyFlag
+  (self, config.application1.registerAppInterfaceParams.appName, config.deviceMAC)
```
Comment (istoimenova) [inline test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua]: @maria-mih, please move parameters on the previous row .

File: test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+function Test:TestStep_PerformAudioPassThru_MandatoryParameters_audioPassThruIcon_SUCCESS()
+  testCasesForPerformAudioPassThru:PerformAudioPassThru_MandatoryParameters_audioPassThruIcon_SUCCESS (self)
```
Comment (istoimenova) [inline test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua]: @maria-mih, as the idea of atomic script is to show what exactly is happening, will be better to declare the TestStep in script, not to call common function.
Calling of common functions is useful as preconditions/postconditions.

Comment (istoimenova) [inline test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua]: @maria-mih , please be sure that RPC PerformAudioPassThru is allowed by policy.

File: test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua (hunk 1 above)
Comment (maria-mih) [inline test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua]: @istoimenova Updated in commit 43f49ed4f7e474fafc736251b7ea5a06ecf1f490

File: test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua (hunk 2 above)
Comment (maria-mih) [inline test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua]: @istoimenova Updated in commit 43f49ed4f7e474fafc736251b7ea5a06ecf1f490

File: test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua (hunk 3 above)
Comment (maria-mih) [inline test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua]: @istoimenova Updated in commit 43f49ed4f7e474fafc736251b7ea5a06ecf1f490

File: test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua (hunk 4 above)
Comment (maria-mih) [inline test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua]: @istoimenova Updated in commit 43f49ed4f7e474fafc736251b7ea5a06ecf1f490

File: test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua (hunk 5 above)
Comment (maria-mih) [inline test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua]: @istoimenova Updated in commit 43f49ed4f7e474fafc736251b7ea5a06ecf1f490

**Comment by maria-mih** [comment]
@istoimenova Script has been updated after first review iteration. Please check on commit 43f49ed4f7e474fafc736251b7ea5a06ecf1f490

File: test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua (hunk 1 above)
Comment (istoimenova) [inline test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua]: @maria-mih, ACCEPTED

File: test_scripts/API/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+-- 1.1. PerformAudioPassThru RPC is allowed by policy
+-- 1.2. Only mandatory parameters are present and within bounds (samplingRate, maxDuration, bitsPerSample, audioType)
+-- as well as AudioPassThruIcon
+-- 1.3. AudioPassThruIcon exists at apps sub-directory of AppStorageFolder (value from ini file)
+-- 2. Performed steps
+-- Send PerformAudioPassThru (audioPassThruIcon, mandatory params) from mobile to SDL and check:
+--
+-- Expected result:
+-- SDL sends UI.PerformAudioPassThru (audioPassThruIcon, other params) to HMI
+-- SDL sends TTS.Speak to HMI
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua]: @maria-mih, as request is sent only with mandatory params, that are applicable for UI, please remove expectations for TTS.

File: test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua (hunk 2 above)
Comment (istoimenova) [inline test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua]: @maria-mih, ACCEPTED.

File: test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua (hunk 3 above)
Comment (istoimenova) [inline test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua]: @maria-mih, ACCEPTED.

File: test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua (hunk 4 above)
Comment (istoimenova) [inline test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua]: @maria-mih, ACCEPTED.

File: test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua (hunk 5 above)
Comment (istoimenova) [inline test_scripts/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua]: @maria-mih, ACCEPTED.

File: test_scripts/API/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+      audioPassThruIcon = 
+      { 
+        value = "icon.png",
+        imageType = "STATIC"      
+  	  }
+    })
+  :Do(function(_,data)
+      self.hmiConnection:SendResponse(data.id, "UI.PerformAudioPassThru", "SUCCESS", {})
+    end)
+
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua]: @maria-mih, please be sure that TTS is not sent
EXPECT_HMICALL("TTS.Speak"):Times(0)

**Comment by maria-mih** [comment]
@istoimenova Please check changes after second review iteration on bea182bb147a8d9576bb5258b82d3bc7425eff57

File: test_scripts/API/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua (hunk 7 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/004_ATF_P_PerformAudioPassThru_SUCCESS_Only_Mandatory_Params_with_audioPassThruIcon.lua]: @istoimenova Updated at bea182bb147a8d9576bb5258b82d3bc7425eff57

**Pull request merged**